src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.Unix.cs

```diff
@@ -193,7 +197,41 @@ public static SafeWaitHandle NewMutex(bool initiallyOwned)
             return safeWaitHandle;
         }
 
-        public static SafeWaitHandle? CreateNamedMutex(bool initiallyOwned, string name, out bool createdNew)
+#if FEATURE_CROSS_PROCESS_MUTEX
+        public static SafeWaitHandle? CreateNamedMutex(bool initiallyOwned, string name, bool isUserScope, out bool createdNew)
+        {
+            try
+            {
+                NamedMutex? namedMutex = NamedMutex.CreateNamedMutex(name, isUserScope, initiallyOwned: initiallyOwned, out createdNew);
+                if (namedMutex == null)
+                {
+                    return null;
+                }
+                SafeWaitHandle safeWaitHandle = NewHandle(namedMutex);
+                return safeWaitHandle;
+            }
+            catch (InvalidSharedMemoryHeaderException ex)
+            {
+                throw new WaitHandleCannotBeOpenedException(SR.Format(SR.Threading_WaitHandleCannotBeOpenedException_InvalidHandle, name), ex);
```

Comment:
What's the advantage of the InvalidSharedMemoryHeaderException exception? Would it be better to throw `WaitHandleCannotBeOpenedException` directly?